homeassistant/components/konnected/config_flow.py

```diff
@@ -0,0 +1,732 @@
+            ATTR_UPNP_MODEL_NAME,
+        )
+
+        _LOGGER.debug(discovery_info)
+
+        try:
+            if discovery_info[ATTR_UPNP_MANUFACTURER] != KONN_MANUFACTURER:
+                return self.async_abort(reason="not_konn_panel")
+
+            if not any(
+                name in discovery_info[ATTR_UPNP_MODEL_NAME]
+                for name in KONN_PANEL_MODEL_NAMES
+            ):
+                _LOGGER.warning(
+                    "Discovered unrecognized Konnected device %s",
+                    discovery_info.get(ATTR_UPNP_MODEL_NAME, "Unknown"),
+                )
+                return self.async_abort(reason="not_konn_panel")
+
+            # extract host/port from ssdp_location
+            netloc = urlparse(discovery_info["ssdp_location"]).netloc.split(":")
+            self.host = self.context["host"] = netloc[0]
+            self.port = netloc[1]
+            self.model = discovery_info[ATTR_UPNP_MODEL_NAME]
+
+        except KeyError:
+            _LOGGER.error("Malformed Konnected SSDP info")
+            return self.async_abort(reason="unknown")
+
+        if any(
+            self.host == flow["context"].get("host")
+            for flow in self._async_in_progress()
+        ):
+            return self.async_abort(reason="already_in_progress")
+
+        if self.host in configured_hosts(self.hass):
+            return self.async_abort(reason="already_configured")
+
+        # brief delay to allow processing of recent status req
+        await asyncio.sleep(0.1)
+
+        # try to obtain the mac address from the device
+        try:
+            self.device_id = self.context["device_id"] = (
+                (await get_status(self.hass, self.host, self.port))
+                .get("mac")
+                .replace(":", "")
+            )
+
+        except CannotConnect:
+            return self.async_abort(reason="cannot_connect")
+
+        if not self.device_id:
+            return self.async_abort(reason="cannot_connect")
+
+        # look for a partially configured instance of this device and hijack it
+        for flow in self.hass.config_entries.flow.async_progress():
+            if flow["handler"] == self.handler and flow["flow_id"] != self.flow_id:
+                if (
+                    self.device_id == flow["context"].get("device_id")
+                    and self.cached_config
+                ):
+                    # steal the config data and abort other flow
+                    # only would happen if the other flow was imported config.yaml w/o host info
+                    _LOGGER.info("Konnected partially imported - completing cfg entry")
+                    config = copy.deepcopy(self.cached_config)
+                    config[CONF_HOST] = self.host
+                    config[CONF_PORT] = self.port
+                    self.hass.config_entries.flow.async_abort(flow["flow_id"])
+                    return await self.async_create_or_update_entry(config)
```

Comment:
A discovery flow should never create an entry without user confirmation.
https://developers.home-assistant.io/docs/en/config_entries_config_flow_handler.html#discovery-steps

It's ok to update an existing config entry with discovery info, matching with unique_id, but not create a new entry without confirmation.

See hue and deconz flows for good examples.

Reply:
👍